src/main/java/uk/ac/ebi/eva/pipeline/jobs/deciders/SkipStepDecider.java

```diff
@@ -29,31 +28,21 @@
 public class SkipStepDecider implements JobExecutionDecider {
     private static final Logger logger = LoggerFactory.getLogger(SkipStepDecider.class);
 
-    private String skipStep;
-    private ObjectMap pipelineOptions;
-
     public static final String SKIP_STEP = "SKIP_STEP";
     public static final String DO_STEP = "DO_STEP";
 
-    /**
-     * @param pipelineOptions ObjectMap that will have a boolean for .get(skipStep), telling whether to skip or not
-     * @param skipStep        name of the key that the user sets to skip a step, e.g. "annotation.create.skip"
-     *                        It's recommended to use pre-defined constants, such as AnnotationJob.SKIP_ANNOT to avoid
-     *                        misspelling mistakes.
-     */
-    public SkipStepDecider(ObjectMap pipelineOptions, String skipStep) {
-        this.skipStep = skipStep;
-        this.pipelineOptions = pipelineOptions;
+    public final String jobParameterName;
+
+    public SkipStepDecider(String jobParameterName){
+        this.jobParameterName = jobParameterName;
     }
 
     @Override
     public FlowExecutionStatus decide(JobExecution jobExecution, StepExecution stepExecution) {
-        if (Boolean.parseBoolean(pipelineOptions.getString(skipStep))) {
-            logger.info("Skipping step because {} is enabled", skipStep);
+        if (Boolean.parseBoolean(jobExecution.getJobParameters().getString(jobParameterName))) {
+            logger.info("Step skipped due to setting {} enabled", jobParameterName);
```

Comment:
`set [something] enabled` is not correct in English. You could say the following:

```
Step skipped due to {} being enabled
````